lib/app/services/network.dart

```diff
@@ -70,122 +68,108 @@ class TooManyRequestsError extends ServerError {
   final Duration timeToWait;
 }
 
-/// A service that offers networking HTTP requests to the backend servers.
-///
-/// It depends on the authentication storage, so if the user's token
-/// is stored there, the requests' headers are automatically enriched with the
-/// access token.
+/// A service that offers networking POST and GET requests to arbitrary
+/// servers.
 @immutable
 class NetworkService {
-  const NetworkService({@required this.apiUrl}) : assert(apiUrl != null);
+  const NetworkService();
+
+  /// Calls the given function and turns various status codes and socket
+  /// exceptions into custom error types like [AuthenticationError] or
+  /// [NoConnectionToServerError].
+  Future<http.Response> _makeCall(Future<http.Response> Function() call) async {
```

Comment:
When merging master into this branch, please make sure to re-add the logging statement and also pass a string of the HTTP method (https://github.com/schul-cloud/schulcloud-flutter/blob/19c0dd1b2cabcbe339f76dd6d04fb15059177e35/lib/app/services/network.dart#L141-L191)